lib/Dialect/FIRRTL/FIRRTLDialect.cpp

```diff
@@ -36,6 +36,28 @@ StringAttr firrtl::getFIRRTLNameAttr(ArrayRef<NamedAttribute> attrs) {
   return StringAttr();
 }
 
+/// If the specified attribute set contains the firrtl.name attribute,
+/// return it and any annotations.
+std::pair<StringAttr, ArrayAttr>
+firrtl::getFIRRTLNameAttrAndAnnotations(ArrayRef<NamedAttribute> attrs) {
```

Comment:
Ah, I see what you mean, yeah it would be better to keep the name handling separate from annotations.  We should keep those in separate encodings.

Reply:
Are you good with putting this in a per-port array like `firrlt.module @Foo { portAnnotations = [{}, {}, ...], portNames = ["", "", ...] }`? I think the consensus is that arg attrs aren't great and we should stay away?

Reply:
Conclusion is to leave this as arg attrs for now because these are expected to be sparse (unlike port names which are mandatory).